### DISCUSSION TITLE
v2.0.5 Patch Release

### DISCUSSION THREAD

**Discussion by Wenn0101** [sparkfun/Arduino_Apollo3#336] [General]
**Release Candidate is up for v2.0.5:**
This is a small release with only a few changes planned. The main motivation for the release is to add the LoRa_Thing_Plus_expLoRaBLE target. In addition I snuck in a few changes to address problems with the last release.

**Changelog**:
-added double quotes around all paths to allow for spaces in windows usernames
-moved -MMD to extra flags, removed it from .cxx-flags, and .c-flags
-Fixed SPI compilation error
-Updates mbed submodule (includes path change for an include for EEPROM)

To test changes, checkout the branch [release-candidate](https://github.com/sparkfun/Arduino_Apollo3/tree/release-candidate) in your apollo3 package location.
I just pushed these changes and I haven't done thorough testing yet. I will keep this thread updated with any additional changes I push.

**Changes since discussion creation:** 
-Commented out Serial1 for nano. (Change pushed to release candidate at ~2100 GMT)

**Comment by jerabaul29** [comment]
- do you want / need some testing already now or wait a bit?
- can you provide detailed explanations of how to test (guess is it not available as an Arduino board yet; how to install from sources then? :) ).

**Reply by paulvha** [reply]
I have downloaded the source with git, but are missing the complete mbed-os and mbed-bridge folder content. I see there are links : mbed-os @ 6b6244e.. on github.  Can you give instruction how to download them.?

**Reply by Wenn0101** [reply]
Yes, it is available to test now.

The testing that I think I would find most useful is:
- If you had an issue with the previous version that is addressed in this release, recreate and test issue is fixed on your system (for example the -MMD build problem on linux)
- Test your projects to see if there are new issues that have occurred in them.

The kinds of issues I will be most interested in are:
- Issues not fixed that are supposed to be fixed in this release
- New issues that are a direct result of this build

Any issues that are not new, should be logged in the typical issues tab in Github, rather than in this discussion.

Right now early testing does need to be done checking out the project via git. Though it could be an interesting idea to have a separate package for pre-release packages in Arduino, this isn't set up.

The easiest way to do this if you have previously installed the arduino core is to:
- navigate to your directory with the latest version
- rename the directory from 2.0.4 to .bckup2.0.4 (or simply delete it and download it later if wanted)
- clone this repo
`git clone https://github.com/sparkfun/Arduino_Apollo3.git`
- checkout release-candidate branch
`git checkout release-candidate`
- update all submodules
`git submodule update --init --recursive`
- rename directory from Arduino_Apollo3 to 2.0.4
- restart Arduino and enjoy.

@paulvha it looks like you will need to update the submodules
`git submodule update --init --recursive`

**Reply by jerabaul29** [reply]
Excellent, will give it a go tonight :) .

**Comment by Wenn0101** [comment]
Thanks again for any help. Let me know if you have any problems.

If you aren't able to fully test, or have other ideas on how to test, that is great! All help and commentary is welcome.

**Comment by paulvha** [comment]
hi

All this worked for me on Ubuntu 20.04. I can compile without errors.. works also when re-compile and then it skips the one that have already been compiled and not modified. Tried 10+ different sketches.

The "new 2.0.4" is MUCH larger due to the .git-folder, but I assume you will remove that before making the release.

Just one request.. it has been on many topics in the forum. An issue and solution with Nano (from KHE)

https://forum.sparkfun.com/viewtopic.php?f=169&t=54315&p=222622&hilit=nano#p222622

Easy fix.... 2 slashes...

Maybe for a next version we should focus to update the variant pin / led  description (pins.cpp and variant.h) for the different boards.

regards,
Paul

________________________________
Van: Wenn0101 <notifications@github.com>
Verzonden: woensdag 27 januari 2021 16:47
Aan: sparkfun/Arduino_Apollo3 <Arduino_Apollo3@noreply.github.com>
CC: paulvha <paulvha@outlook.com>; Mention <mention@noreply.github.com>
Onderwerp: Re: [sparkfun/Arduino_Apollo3] v2.0.5 Patch Release (#336)


Yes, it is available to test now.

The testing that I think I would find most useful is:

  *   If you had an issue with the previous version that is addressed in this release, recreate and test issue is fixed on your system (for example the -MMD build problem on linux)
  *   Test your projects to see if there are new issues that have occurred in them.

The kinds of issues I will be most interested in are:

  *   Issues not fixed that are supposed to be fixed in this release
  *   New issues that are a direct result of this build

Any issues that are not new, should be logged in the typical issues tab in Github, rather than in this discussion.

Right now early testing does need to be done checking out the project via git. Though it could be an interesting idea to have a separate package for pre-release packages in Arduino, this isn't set up.

The easiest way to do this if you have previously installed the arduino core is to:

  *   navigate to you directory with the latest version
  *   rename the directory from 2.0.4 to .bckup2.0.4 (or simply delete it and download it later if wanted)
  *   clone this repo
git clone https://github.com/sparkfun/Arduino_Apollo3.git
  *   checkout release-candidate branch
git checkout release-candidate
  *   update all submodules
git submodule update --init --recursive
  *   rename directory from Arduino_Apollo3 to 2.0.4
  *   restart Arduino and enjoy.

@paulvha<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpaulvha&data=04%7C01%7C%7C4c6ebecfdd7e4b38018f08d8c2daeb88%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637473592787915968%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=A2ZvQqVMs4%2FSxfB2nRp9QJrSTyBOiIIkR9Ru4%2BxisyY%3D&reserved=0> it looks like you will need to update the submodules
git submodule update --init --recursive

**Reply by Wenn0101** [reply]
Suggestion taken. Commented out Serial1. Changes pushed to dev, and new updated version is on the release-candidate.

.git folder will not be in release package

**Reply by jerabaul29** [reply]
Works fine here on Ubuntu 18.04. Only tested a few sketches, but things look good.

**Comment by Wenn0101** [comment]
v2.0.5 is now live!
Let me know if you have any problems with the released version!

**Comment by paulvha** [comment]
Just tried a new sketch and somehow analogReadVCCDiv3() together with many others like am_hal_pwrctrl_periph_disable(AM_HAL_PWRCTRL_PERIPH_BLEL); can not be found... not sure why.. it is in 2.0.5 but maybe a link in an include file is missing.. don't know yet....

regards,
Paul

________________________________
Van: Wenn0101 <notifications@github.com>
Verzonden: maandag 1 februari 2021 21:00
Aan: sparkfun/Arduino_Apollo3 <Arduino_Apollo3@noreply.github.com>
CC: paulvha <paulvha@outlook.com>; Mention <mention@noreply.github.com>
Onderwerp: Re: [sparkfun/Arduino_Apollo3] v2.0.5 Patch Release (#336)


v2.0.5 is now live!
Let me know if you have any problems with the released version!

**Comment by paulvha** [comment]
correction. 
The update did not work, but a re-install did. After selecting the right board ( ATP) compiled worked without error messages

regards,
Paul

**Reply by Wenn0101** [reply]
Good to hear!
